neo/rawio/openephysbinaryrawio.py

```diff
@@ -192,10 +224,44 @@ def _parse_header(self):
                             "SYNC channel is not present in the recording. " "Set load_sync_channel to False"
                         )
 
-                    if has_sync_trace and not self.load_sync_channel:
-                        self._stream_buffer_slice[stream_id] = slice(None, -1)
+                    # Check if ADC and non-ADC channels are contiguous
+                    is_channel_adc = ["ADC" in ch["channel_name"] for ch in info["channels"]]
+                    first_adc_index = is_channel_adc.index(True) if any(is_channel_adc) else len(is_channel_adc)
+                    non_adc_channels_after_adc_channels = [not flag for flag in is_channel_adc[first_adc_index:]]
+                    if any(non_adc_channels_after_adc_channels):
+                        raise ValueError(
+                            "Interleaved ADC and non-ADC channels are not supported. "
+                            "ADC channels must be contiguous. Open an issue in python-neo to request this feature."
+                        )
+
+                    # Find sync channel and verify it's the last channel
+                    sync_index = next(
+                        (index for index, ch in enumerate(info["channels"]) if ch["channel_name"].endswith("_SYNC")),
+                        None,
+                    )
+                    if sync_index is not None and sync_index != num_channels - 1:
+                        raise ValueError(
+                            "SYNC channel must be the last channel in the buffer. Open an issue in python-neo to request this feature."
+                        )
+
+                    num_neural_channels = sum(1 for ch_info in info["channels"] if "ADC" not in ch_info["channel_name"])
+                    num_non_neural_channels = sum(1 for ch_info in info["channels"] if "ADC" in ch_info["channel_name"])
```

Comment:
```suggestion
                    num_neural_channels = sum("ADC" not in ch_info["channel_name"] for ch_info in info["channels"])
                    num_non_neural_channels = sum("ADC" in ch_info["channel_name"] for ch_info in info["channels"])
```
easier to read?

Reply:
It is easier to read than mine. My fault. But now that I think about it, it would be easier to get a list and then use length.

Reply:
I just defined first the list and then used len().

Trying to reduce number of lines is a foolish endeavor. Un gusto culposo.